presto-main/src/test/java/com/facebook/presto/execution/resourceGroups/TestResourceGroups.java

```diff
@@ -332,10 +338,43 @@ public void testSoftCpuLimit()
         assertEquals(query3.getState(), RUNNING);
     }
 
+    @Test(timeOut = 10_000)
+    public void testPerWorkerQueryLimit()
```

Comment:
add a unit test to verify if the resource group tree has root -> group1 and root -> group2 hierarchy, then also it works correctly.